lib/jsonwp-proxy/proxy.js

```diff
@@ -132,91 +145,79 @@ class JWProxy {
-      // `res.body` might be really big
+      const {data, status, headers} = await this.request(reqOpts);
+      // `data` might be really big
       // Be careful while handling it to avoid memory leaks
-      const resBodyObj = util.safeJsonParse(res.body);
-      if (!_.isPlainObject(resBodyObj)) {
+      if (!_.isPlainObject(data)) {
         // The response should be a valid JSON object
         // If it cannot be coerced to an object then the response is wrong
-        throwProxyError(res.body);
+        throwProxyError(data);
       }
-      log.debug(`Got response with status ${res.statusCode}: ${truncateBody(res.body)}`);
+      log.debug(`Got response with status ${status}: ${truncateBody(data)}`);
       isResponseLogged = true;
       const isSessionCreationRequest = /\/session$/.test(url) && method === 'POST';
       if (isSessionCreationRequest) {
-        if (res.statusCode === 200) {
-          this.sessionId = resBodyObj.sessionId || (resBodyObj.value || {}).sessionId;
+        if (status === 200) {
+          this.sessionId = data.sessionId || (data.value || {}).sessionId;
         }
-        this.downstreamProtocol = this.getProtocolFromResBody(resBodyObj);
+        this.downstreamProtocol = this.getProtocolFromResBody(data);
         log.info(`Determined the downstream protocol as '${this.downstreamProtocol}'`);
       }
-      if (res.statusCode < 400 && _.has(resBodyObj, 'status') && parseInt(resBodyObj.status, 10) !== 0) {
+      if (status < 400 && _.has(data, 'status') && parseInt(data.status, 10) !== 0) {
         // Some servers, like chromedriver may return response code 200 for non-zero JSONWP statuses
-        throwProxyError(resBodyObj);
+        throwProxyError(data);
       }
-      return [res, resBodyObj];
+      const res = {statusCode: status, headers, body: data};
+      return [res, data];
     } catch (e) {
       // We only consider an error unexpected if this was not
       // an async request module error or if the response cannot be cast to
       // a valid JSON
-      if (util.hasValue(e.error)) {
+      if (util.hasValue(e.response)) {
         if (!isResponseLogged) {
-          const error = truncateBody(e.error);
-          log.info(util.hasValue(e.statusCode)
-            ? `Got response with status ${e.statusCode}: ${error}`
+          const error = truncateBody(e.response.data);
+          log.info(util.hasValue(e.response.status)
+            ? `Got response with status ${e.response.status}: ${error}`
             : `Got response with unknown status: ${error}`);
         }
       } else {
         log.warn(e.message);
         log.debug(e.stack);
       }
-      throw new errors.ProxyRequestError(`Could not proxy command to remote server. ` +
-        `Original error: ${e.message}`, e.error, e.statusCode);
+      throw new errors.ProxyRequestError(`Could not proxy command to the remote server. ` +
+        `Original error: ${e.message}`, e.response?.data, e.response?.status);
```

Comment:
oh we can use null coalescing now? yay!

Reply:
this is available since gulp plugins 5.2.0